server/test/services/applicant/ApplicantServiceTest.java

```diff
@@ -951,13 +957,52 @@ public void submitApplication_failsWithApplicationNotEligibleException() {
                         applicant.id,
                         programDefinition.id(),
                         trustedIntermediaryProfile,
-                        /* eligibilityFeatureEnabled= */ true)
+                        /* eligibilityFeatureEnabled= */ true,
```

Comment:
Should we have test cases where one is true and the other is false? Or is that not a possible condition?

Reply:
There were no existing tests for the eligibility flag, but sure added a couple of tests to make sure both flags are being respected.